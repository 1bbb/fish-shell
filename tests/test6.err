funcevents: Expected one function name as argument, got 0
funcevents: Expected one function name as argument, got 2
