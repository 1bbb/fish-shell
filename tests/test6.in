#
#Test basic funcevents behavior
#

function testfn --on-event=foo
	echo bar.$FOOIDX
end

function show_idx_change --on-variable=FOOIDX
	echo FOOIDX is now $FOOIDX
end

set FOOIDX 1

# should print 'bar.1'
emit foo

# should print 'testfn'
funcevents -e foo

# no function name - should give an error
funcevents -r -e foo

# this time it should work
funcevents -r -e foo testfn

# should no longer print anything
emit foo

# too many functions - should give error
funcevents -e foo testfn show_idx_change

funcevents -e foo testfn
set FOOIDX 2

# should print 'bar.2'
emit foo

# show descriptions of current subscriptions
funcevents testfn

