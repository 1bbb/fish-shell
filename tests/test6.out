FOOIDX is now 1
bar.1
testfn
FOOIDX is now 2
bar.2
handler for generic event “foo”
